pkg/machine/ports.go

```diff
@@ -0,0 +1,193 @@
+package machine
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/containers/storage/pkg/ioutils"
+	"github.com/containers/storage/pkg/lockfile"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	portAllocFileName = "port-alloc.dat"
+	portLockFileName  = "port-alloc.lck"
+)
+
+func AllocateMachinePort() (int, error) {
```

Comment:
This function also need comments, it was not obvious to me why we want to store ports for all machines?
I assume you want to avoid conflicts if multiple machine are used at the same time?

Reply:
@Luap99 That's right it's to ensure that machines can never conflict with each other when running at the same time. While they could just detect and reassign without coordination, there is a very narrow window between the detect bind and the actual ssh bind where a machine start could theoretically fail to properly adapt. It also could lead to a bit of a prolonged dance where ports keep changing until the total set is conflict free (A takes C port to avoid conflict with B, C takes Bs port to avoid conflict with A, etc) . By ensuring we assign conflict free with other machines those issues are prevented.  Good point on comments, I will add more